### DISCUSSION TITLE
Blocking network urls in uc mode

### DISCUSSION THREAD

**Discussion by iohansson** [seleniumbase/SeleniumBase#2580] [Q&A] [answered]
I'm blocking some urls with `execute_cdp_cmd` and it seems that it doesn't work. I suspect that this is because of the reconnect. `Network.setBlockedURLs` and `Network.enable` approach works with `undetected_chromedriver` but not `seleniumbase`.
What would be a proper way of blocking resources? The goal is to save traffic when using a proxy.

**Comment by mdmintz** [comment, marked as answer]
The methods like `uc_open_with_tab(url)` and `uc_open_with_reconnect(url, reconnect_time)` open URLs in a new tab. However, `execute_cdp_cmd()` only affects the current tab. You can get around that by using `uc_open(url)` after you've called `execute_cdp_cmd()` (or if you refresh the page).

**Reply by iohansson** [reply]
Hmm, that was my first thought to try that :) And it didn't help, could you check if it works for you currently?

**Reply by mdmintz** [reply]
```python
from seleniumbase import BaseCase
BaseCase.main(__name__, __file__, "--uc")


class TestGeolocation(BaseCase):
    def tearDown(self):
        self.save_teardown_screenshot()  # If test fails, or if "--screenshot"
        if self.is_chromium() and not self._multithreaded:
            # Reset Permissions and GeolocationOverride
            try:
                self.open("about:blank")
                self.execute_cdp_cmd("Emulation.setGeolocationOverride", {})
                self.execute_cdp_cmd("Browser.resetPermissions", {})
            except Exception:
                pass
        super().tearDown()

    def test_geolocation(self):
        self.open("about:blank")
        if self._multithreaded:
            self.skip("Skipping test in multi-threaded mode.")
        if not self.is_chromium():
            print("\n* execute_cdp_cmd() is only for Chromium browsers")
            self.skip("execute_cdp_cmd() is only for Chromium browsers")
        self.driver.uc_open("https://www.openstreetmap.org/")
        self.execute_cdp_cmd(
            "Browser.grantPermissions",
            {
                "origin": "https://www.openstreetmap.org/",
                "permissions": ["geolocation"],
            },
        )
        self.execute_cdp_cmd(
            "Emulation.setGeolocationOverride",
            {
                "latitude": 48.87645,
                "longitude": 2.26340,
                "accuracy": 100,
            },
        )
        self.refresh()
        self.click("span.geolocate")
        self.assert_url_contains("48.87645/2.26340")
        self.save_screenshot_to_logs()
        if self.headed:
            self.sleep(2.5)
```

**Reply by iohansson** [reply]
Thanks! I wasn't using the BaseCase, just the driver, do I have to? Is `self.execute_cdp_cmd` just a proxy for the same method on the driver or not?

**Reply by mdmintz** [reply]
The `BaseCase` methods can also be reached from the `SB()` / `sb` format. Eg:

```python
from seleniumbase import SB

with SB() as sb:
    sb.open("seleniumbase.io/simple/login")
    sb.type("#username", "demo_user")
    sb.type("#password", "secret_pass")
    sb.click('a:contains("Sign in")')
    sb.assert_exact_text("Welcome!", "h1")
    sb.assert_element("img#image1")
    sb.highlight("#image1")
    sb.click_link("Sign out")
    sb.assert_text("signed out", "#top_message")
```

The `SB()` methods are more advanced than the `Driver()` methods. (The `Driver()` methods are basically what comes with Selenium, with a few extra methods added in for simplification.) You can also reach the `Driver()` methods from the `SB()` format by calling `sb.driver`. For some methods that can be found in both formats, the `SB()` version is often more powerful than the `Driver()` version.

**Reply by iohansson** [reply]
ok, I'll give it a shot, thank you!
